x/oracle/keeper/band_oracle_test.go

```diff
@@ -132,3 +133,144 @@ func TestBandCallDataRecord(t *testing.T) {
 	record = app.OracleKeeper.GetBandCallDataRecord(ctx, 1)
 	require.Nil(t, record)
 }
+
+func TestGetPrice(t *testing.T) {
+	app := app.Setup(t, false)
+	ctx := app.BaseApp.NewContextLegacy(false, tmproto.Header{Height: 1, ChainID: "3", Time: time.Unix(1618997040, 0)})
+
+	// Setup test data
+	bandPriceStateATOM := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(10),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(10), 1),
+	}
+	bandPriceStateUSD := &types.BandPriceState{
+		Symbol:      "USD",
+		Rate:        math.NewInt(1),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(1), 1),
+	}
+	bandPriceStateNOM := &types.BandPriceState{
+		Symbol:      "NOM",
+		Rate:        math.NewInt(2),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(2), 1),
+	}
+	invalidPriceStateATOM := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(0), // Invalid base rate
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(0), 1),
+	}
+
+	// Create variables for expected prices
+	expectedPrice10 := math.LegacyNewDec(10)
+	expectedPrice05 := math.LegacyNewDec(5)   // For ATOM/NOM (10/2)
+	expectedPrice01 := math.LegacyNewDec(1).Quo(math.LegacyNewDec(10)) // 0.1
+
+	tests := []struct {
+		name            string
+		baseSymbol      string
+		quoteSymbol     string
+		basePriceState  *types.BandPriceState
+		quotePriceState *types.BandPriceState
+		expectedPrice   *math.LegacyDec
+		expectNil       bool
+	}{
+		// Fail cases first
```

Comment:
It looks like we don't have the fail case, instead we should separate into 2 cases return nil and return price @hungdinh82